### DISCUSSION TITLE
Weird glitchy frame when using -page on gifs

### DISCUSSION THREAD

**Discussion by EPPI-c** [ImageMagick/ImageMagick#6895] [Help] [answered]
### ImageMagick version

7.1.1-21 

### Operating system, version and so on

Fedora release 38 kernel: 6.5.12-200.fc38.x86_64

### Description

```sh
#!/bin/sh
owidth=$(identify -ping -format "%w" "$1"[0])
oheight=$(identify -ping -format "%h" "$1"[0])
mold=$(($oheight*1/10))
width=$(($owidth + $mold*2))
height=$(($oheight + $mold*2))
convert -gravity center \( \( -size $width\x$height -raise $mold\x$mold xc:tan4 \) -size $owidth\x$oheight xc:white -composite \) -page +$mold\+$mold "$1" -coalesce -delete 0 moldure.png
```
input:
![kusuriya-no-hitorigoto-mao-mao](https://github.com/ImageMagick/ImageMagick/assets/109323250/19b69d00-5f1a-4dfc-9bd5-c0f2d6e16514)

output:
![moldure](https://github.com/ImageMagick/ImageMagick/assets/109323250/52a70df4-0cd0-42e2-ae60-258bf668e601)

for some reason when I use -page or -repage it generates this frame
![moldure-5](https://github.com/ImageMagick/ImageMagick/assets/109323250/8480106e-9f4f-4f88-b66e-c33dedf0bbe2)

When I remove the -page/-repage it generates the gif just fine
![moldure](https://github.com/ImageMagick/ImageMagick/assets/109323250/a7fa3cf1-c89e-42b5-b863-8e3514edb1ae)

Am I doing something wrong?

**Comment by snibgo** [comment, marked as answer]
As you have IM v7, I suggest using `magick` instead of `convert`.

For the output, do you want an animated GIF, or a single image PNG, or multiple PNG files?

You are applying `-coalesce` to all the images, which are the frames from your GIF, and the "xc:" image. This seems like a bad idea. You should `-coalesce` only the GIF frames. Then apply the border or framing effect you want.

**Reply by EPPI-c** [reply]
Thanks! The suggestion to coalesce only the gif did the trick. From my testing I suspect the weird glitched frames were happening when I used -page or -repage before coalescing the gif. I still find it a bit odd because I moved them all to the same position so in my understanding there shouldn't be any problem.

the final script using magick command:
```sh
#!/bin/sh
owidth=$(identify -ping -format "%w" "$1"[0])
oheight=$(identify -ping -format "%h" "$1"[0])
mold=$(($oheight*1/10))
width=$(($owidth + $mold*2))
height=$(($oheight + $mold*2))

magick \( \( -size $width\x$height xc:tan4 -raise $mold\x$mold \) -size $owidth\x$oheight xc:white -gravity center -composite \) \( "$1" -coalesce -repage +$mold+$mold \) -coalesce -layers RemoveZero moldure.gif
```
it also worked when using the older command:
```sh
convert -gravity center \( \( -size $width\x$height -raise $mold\x$mold xc:tan4 \) -size $owidth\x$oheight xc:white -composite \) \( "$1" -coalesce -repage +$mold\+$mold \) -coalesce -layers RemoveZero moldure.gif
```
I had to coalesce again after that to remove the first frame which causes an annoying flicker in the image.
Annoying flicker:
![moldure](https://github.com/ImageMagick/ImageMagick/assets/109323250/1e9d8b73-8a46-4f51-8278-203981ed9fef)

when I didn't coalesce it ended up removing the whole frame:
![moldure](https://github.com/ImageMagick/ImageMagick/assets/109323250/608f016f-6c51-4f47-af0e-74a12eceb673)

Coalescing twice seems a bit of a waste but I don't know if there is a better way.
![moldure](https://github.com/ImageMagick/ImageMagick/assets/109323250/11cad1c8-d772-4a9a-bb98-f4756ba22239)

I have some other questions if you could help me with those as well I'd be very happy.
- you mentioned I should use the magick command, is there some place I can see the benefits from the new syntax over the old one, I'd love to understand better what made the developers change syntax, and what the differences are between the old and the new commands.
- while trying to remove the flicker effect I stumbled upon this `-layer RemoveZero` option, previously I was using -delete 0 as I saw in some example in the documentation. Is there a difference between both or are they synonyms?
- are you aware of any other way I could have solved the flicker problem without having to coalesce twice?

**Reply by EPPI-c** [reply]
Ok apparently I could just have used -layers composite, removing the duplicate coalesce and the need to repage and the need to remove the first frame.
```sh
magick \( \( -size $width\x$height xc:tan4 -raise $mold\x$mold \) -size $owidth\x$oheight xc:white -gravity center -composite \) NULL: \( "$1" -coalesce \) -layers composite moldure.gif
```
